lib/settings.js

```diff
@@ -1,135 +1,126 @@
-var fs = require('fs')
+var Promise = require('bluebird')
     , path = require('path')
-    , ini = require('config-ini')
+    , fs = Promise.promisifyAll(require('fs'))
+    , ini = Promise.promisifyAll(require('config-ini'))
     , expandTilde = require('expand-tilde')
-    , deasync = require('deasync')
     , settings = module.exports;
 
-var indicorc = function() {
+var loadIndicorc = function() {
+  return new Promise(function(resolve, reject){
     /*
     Search for valid indicorc files to use as configuration
     Load files in order of precendence if any are found
     */
+
     var localPath = path.resolve(process.cwd(), './.indicorc');
     var globalPath = expandTilde('~/.indicorc');
     var paths = [globalPath, localPath];
-    
-    var valid_paths = [];
-    for (i in paths) {
+    var validPaths = [];
+
+    for (var i = path.length - 1; i >= 0; i--) {
         if (fs.existsSync(paths[i])) {
-            valid_paths.push(paths[i]);
+            validPaths.push(paths[i]);
         }
     }
 
-    var done = false;
-    if (valid_paths.length > 0) {
-        ini.load(valid_paths, function(err, config) {
-            if (err) {
-                throw new Error(err);
-            } else {
-                done = true;
-            }
-        });
-
-        // block until the config file has been loaded
-        while (!done) {
-            deasync.runLoopOnce()
-        }
-
-        return ini
+    if (paths.length < 1) {
+        reject('Could not find .indicorc and no local path was given');
     }
+
+    resolve(validPaths);
+  })
+  .then(ini.load);
 }
 
-var indicorcFile = indicorc();
+var indicorc = loadIndicorc().catch(function(err){
+  console.warn(err);
```

Comment:
I added this not understanding that it wasn't required at first. Instead there should be a warning if there isn't and api key found, which I'm not sure is actually done. Will find and comment on the spot that I think it should occur if it can't find an apikey from any source.